guides/hotwire-turbo-micronaut-chat/complete/java/src/test/java/example/micronaut/controllers/MessagesControllerTest.java

```diff
@@ -0,0 +1,92 @@
+package example.micronaut.controllers;
+
+import example.micronaut.TestUtils;
+import io.micronaut.context.annotation.Property;
+import io.micronaut.core.io.ResourceLoader;
+import io.micronaut.core.util.StringUtils;
+import example.micronaut.entities.Room;
+import example.micronaut.repositories.MessageRepository;
+import example.micronaut.repositories.RoomRepository;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.client.BlockingHttpClient;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.uri.UriBuilder;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Property(name = "micronaut.http.client.follow-redirects", value = StringUtils.FALSE)
+@MicronautTest(transactional = false)
```

Comment:
Because I save things in the database within the test methods and I need those things to be committed to the database before other request hits the server. E.g. 

In the following code: 

```
   @Test
     void saveAMessage() {
         BlockingHttpClient client = httpClient.toBlocking();
         Room roomA = roomRepository.save("Room A");

         URI uri = UriBuilder.of("/rooms").path("" + roomA.getId()).path("messages").build();
         HttpRequest<?> request = HttpRequest.POST(uri, Collections.singletonMap("content", "Hola"))
                 .contentType(MediaType.APPLICATION_FORM_URLENCODED);
         HttpResponse<?> response = client.exchange(request);
         assertEquals(HttpStatus.SEE_OTHER, response.status());
         String location = response.getHeaders().get(HttpHeaders.LOCATION);
         assertNotNull(location);
         assertTrue(location.startsWith("/rooms/"));

         messageRepository.deleteAll();
         roomRepository.deleteById(roomA.getId());
     }
```

I need this room 

```
         Room roomA = roomRepository.save("Room A");
```

to be saved before this request hits: 

```
URI uri = UriBuilder.of("/rooms").path("" + roomA.getId()).path("messages").build();
         HttpRequest<?> request = HttpRequest.POST(uri, Collections.singletonMap("content", "Hola"))
                 .contentType(MediaType.APPLICATION_FORM_URLENCODED);
```